src/js/components/app.test.js

```diff
@@ -58,16 +61,25 @@ describe('App Component', () => {
   });
 
   it('works as intended', async () => {
-    window.localStorage.getItem.mockReturnValueOnce('false');
-    Object.defineProperty(window.document, 'cookie', {
-      writable: true,
-      value: 'JWT=omnomnom'
+    const store = mockStore({
+      ...state,
+      users: {
+        ...state.users,
+        currentUser: 'notNull'
+      }
     });
-    render(
+    window.localStorage.getItem.mockReturnValueOnce('false');
+    const ui = (
       <Provider store={store}>
         <App />
       </Provider>
     );
-    await act(async () => jest.advanceTimersByTime(900500));
+    const { rerender } = render(ui);
+    await act(async () => jest.advanceTimersByTime(timeout + 500));
```

Comment:
in general this seems to a a long wait. two questions:
* why are we waiting that long?
* why are we adding a very small fraction of that wait here?

Reply:
the `timeout` is what is used in the UI to do the auto logout when a user is idle - instead of waiting for that time to pass, the `advanceTimersByTime` forwards the time in the test run by the defined amount, so we skip ahead by just a bit more than the timeout and cause the idle check to trigger => this should get us logged out

Reply:
right! I was missing some parts. thanks.